### PR TITLE
Revert saving the dataloader and result collection by default

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Partially reverts #10784 so the dataloader state and result metrics are not collected with the checkpoint

This sidesteps an error reported by @ananthsub 

### Does your PR introduce any breaking changes? If yes, please list them.

None, this is experimental.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/loops/epoch/training_epoch_loop.py [hunk 1]
```diff
@@ -264,7 +264,9 @@ def on_save_checkpoint(self) -> Dict:
         state_dict = super().on_save_checkpoint()
 
         if (
-            self.trainer.train_dataloader is None
+            self.trainer is None
```
[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/loops/epoch/training_epoch_loop.py:267]: Yes for some simple cases but we would need to extend the coverage before reverting this PR in the future

**Review by ethanwharris — APPROVED** [review]
